src/main/java/GameOfLife.java

```diff
@@ -0,0 +1,69 @@
+public class GameOfLife {
+    private int rows;
+    private int cols;
+    private int[][] grid;
+
+    public GameOfLife(int rows, int cols) {
+        this.rows = rows;
+        this.cols = cols;
+        this.grid = new int[rows][cols];
+    }
+
+    public void setCell(int row, int col, int value) {
+        grid[row][col] = value;
+    }
+
+    public void setGrid(int[][] newGrid) {
+        this.grid = newGrid;
+        this.rows = newGrid.length;
+        this.cols = newGrid[0].length;
+    }
+
+    public void nextGen() {
+        int[][] newGrid = new int[rows][cols];
+
+        for (int row = 0; row < rows; row++) {
+            for (int col = 0; col < cols; col++) {
+                int aliveNeighbors = countAliveNeighbors(row, col);
+
+                if (grid[row][col] == 1) {
+                    newGrid[row][col] = (aliveNeighbors == 2 || aliveNeighbors == 3) ? 1 : 0;
+                } else {
+                    newGrid[row][col] = (aliveNeighbors == 3) ? 1 : 0;
+                }
+            }
+        }
+
+        grid = newGrid;
+    }
+
+    private int countAliveNeighbors(int row, int col) {
+        int count = 0;
+        for (int i = -1; i <= 1; i++) {
+            for (int j = -1; j <= 1; j++) {
+                if (i == 0 && j == 0) {
+                    continue;
+                }
+                int newRow = row + i;
+                int newCol = col + j;
+
+                if (newRow >= 0 && newRow < rows && newCol >= 0 && newCol < cols) {
+                    count += grid[newRow][newCol];
```

Comment:
`count++`

Reply:
that increments all of the neighbors whether they are alive or not?

Reply:
shouldn't it only be if the neighbor is alive? otherwise the test fails because the output is incorrect

Reply:
Nevermind